jumanji/environments/swarms/search_and_rescue/env.py

```diff
@@ -0,0 +1,390 @@
+
+        Args:
+            key: Random key used to reset the environment.
+
+        Returns:
+            state: Initial environment state.
+            timestep: TimeStep with individual search agent views.
+        """
+        state = self.generator(key, self.searcher_params)
+        timestep = restart(observation=self._state_to_observation(state))
+        return state, timestep
+
+    def step(self, state: State, actions: chex.Array) -> Tuple[State, TimeStep[Observation]]:
+        """Environment update.
+
+        Update searcher velocities and consequently their positions,
+        mark found targets, and generate rewards and local observations.
+
+        Args:
+            state: Environment state.
+            actions: Arrays of searcher steering actions.
+
+        Returns:
+            state: Updated searcher and target positions and velocities.
+            timestep: Transition timestep with individual agent local observations.
+        """
+        searchers = update_state(state.key, self.searcher_params, state.searchers, actions)
+        key, target_key = jax.random.split(state.key, num=2)
+        # Ensure target positions are wrapped
+        target_pos = self._target_dynamics(target_key, state.targets.pos) % 1.0
+        # Grant searchers rewards if in range and not already detected
+        # spatial maps the has_found_target function over all pair of targets and
+        # searchers within range of each other and sums rewards per agent.
+        rewards = spatial(
+            utils.has_found_target,
+            reduction=jnp.add,
+            default=0.0,
+            i_range=self.target_contact_range,
+        )(
+            key,
+            self.searcher_params.view_angle,
+            searchers,
+            state.targets,
+            pos=searchers.pos,
+            pos_b=target_pos,
+        )
+        # Mark targets as found if with contact range and view angle of a searcher
+        # spatial maps the has_been_found function over all pair of targets and
+        # searchers within range of each other
+        targets_found = spatial(
+            utils.has_been_found,
+            reduction=jnp.logical_or,
+            default=False,
+            i_range=self.target_contact_range,
+        )(
+            key,
+            self.searcher_params.view_angle,
+            state.targets.pos,
+            searchers,
+            pos=target_pos,
+            pos_b=searchers.pos,
+        )
+        # Targets need to remain found if they already have been
+        targets_found = jnp.logical_or(targets_found, state.targets.found)
+        state = State(
+            searchers=searchers,
+            targets=TargetState(pos=target_pos, found=targets_found),
+            key=key,
+            step=state.step + 1,
+        )
+        observation = self._state_to_observation(state)
+        timestep = jax.lax.cond(
+            state.step >= self.max_steps | jnp.all(targets_found),
```

Comment:
In general it's good to be quite overly careful with the brackets for jax conditions, I think this may have done an or with `max_steps` and `targets_found`
```suggestion
            (state.step >= self.max_steps) | jnp.all(targets_found),
```